### DISCUSSION TITLE
Change workflows and branches rules to allow merging PRs with linting errors.

### DISCUSSION THREAD

**Discussion by josecelano** [torrust/torrust-tracker#546] [Proposals]
Since we are using always the latest Rust and Clipply versions in our workflows very often we can't merge PRs because some new listing errors are not related to what has been changed in the PR.

For example https://github.com/torrust/torrust-tracker/pull/522

That could be pretty annoying because you have to fix something you have not done. And it can discourage contributors from opening or finishing PRs.

We have been discussing it and one of the proposals is

### Proposal 1

- Run format and linter checks only for new or changed files in the PR (git diff).
- Create a new workflow (like the `dependabot`) which runs periodic formatting checks with the latest Rsut and Clippy version. Ideally, it could even create new issues automatically when a new formatting error appears.

This has also another advantage, you do not need to change the rules to merge critical PRs like the ones created to fix bugs or security issues.

cc @da2ce7 @WarmBeer @mario-nt @grmbyrn @cgbosse

**Comment by cgbosse** [comment]
I would go with the proposal 1.

**Comment by mario-nt** [comment]
I also agree that only checking the new code that is going to be implemented is the way to go. Does not make sense to have a contributor working on errors coming from something that person hasn't worked on, as it will create a lot of frustration and extra unwanted work.

As Jose said, we should only check the new or modified files, and have some sort of automation for any errors that arise in other parts of the code.
